compiler/src/iree/compiler/Codegen/LLVMCPU/Passes.cpp

```diff
@@ -685,6 +686,7 @@ static void addLowerToLLVMPasses(OpPassManager &passManager) {
     passManager.addNestedPass<func::FuncOp>(
         createLLVMCPUEmitVectorizationRemarksPass());
   }
+  passManager.addPass(IREE::Util::createPromoteArithBF16ToF32Pass());
```

Comment:
I am wondering if this is enough. Arent the operations in the body `linalg` operations with `bf16` types? I am wondering why this is different compared to the `i1` -> `i8` handling?

Reply:
Do we/could we add a test that verifies this?

Reply:
We may be able to do some similar tricks for `i1` packing as well. I imagine `exti` could have a similar conversion where we extract only the bits needed, though the loading semantics would need to be adjusted.

Reply:
ok I see... well it makes me a bit concerned that we have two places where we are adjusting types... but it is true, one is at buffer level and another is at register level....